tests/test_scheduler.py

```
@@ -0,0 +1,527 @@
+            loss.backward()
+            model.step()
+        
+        pred_tree0 = self._get_tree_predictions(model, 'cpu', 0, 1)
+        pred_tree1 = self._get_tree_predictions(model, 'cpu', 1, 2)
+        
+        tree0_contribution = np.abs(pred_tree0).mean()
+        tree1_contribution = np.abs(pred_tree1).mean()
+        
+        # With constant lr=1.0, both trees should have same scaling
+        # (though raw leaf values differ based on residuals)
+        print(f"\nCPU Constant Scheduler Test:")
```

Ruff (0.14.14)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)